packages/itwin/grouping-mapping-widget/src/components/Mappings/hooks/useRunExtraction.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { useExtractionClient } from "../../context/ExtractionClientContext";
+import { useMutation } from "@tanstack/react-query";
+import type { GroupingMappingApiConfig } from "../../context/GroupingApiConfigContext";
+import type { ExtractionRequestMapping, ExtractionRunRequest, Mapping } from "@itwin/insights-client";
+import { useState } from "react";
+import { useExtractionStateJobContext } from "../../context/ExtractionStateJobContext";
+
+export interface FetchExtractionStatesProps extends GroupingMappingApiConfig {
+  jobId?: string;
+}
+
+export const useRunExtraction = ({
+  iModelId,
+  getAccessToken,
+}: FetchExtractionStatesProps) => {
+  const extractionClient = useExtractionClient();
+  const [isJobStarted, setIsJobStarted] = useState<boolean>(false);
+  const { mappingIdJobInfo, setMappingIdJobInfo } = useExtractionStateJobContext();
+
+  const { mutateAsync: runExtraction, isLoading: isRunExtractionLoading, isSuccess: isRunExtractionSuccess } = useMutation({
+    mutationKey: ["runExtraction"],
+    mutationFn: async (mappings: Mapping[]) => {
+      const accessToken = await getAccessToken();
+      const mappingIds: ExtractionRequestMapping[] = mappings.length > 0 ? mappings.map((mapping) => { return { id: mapping.id }; }) : [];
+      const extractionRequest: ExtractionRunRequest | undefined = {
```

Comment:
What's the point of having `| undefined` if it is `const` and initialized with a non-undefined value?

Reply:
Removed.